src/core/constraints.c

```diff
@@ -1345,9 +1347,8 @@ constrain_aspect_ratio (MetaWindow         *window,
                                                       new_width, new_height,
                                                       &best_width, &best_height);
 
-      /* Yeah, I suck for doing implicit rounding -- sue me */
-      new_width  = best_width;
-      new_height = best_height;
+      new_width  = round (best_width);
+      new_height = round (best_height);
```

Comment:
As it's cast to int anyway, why not use this simple version that avoids using the C99 `round()` that introduce a new check that even failed at some point?
```c
      new_width  = (int) (best_width + 0.5);
      new_height = (int) (best_height + 0.5);
```
The computation result is the same (or at least virtually so, I didn't check the standard to see if that can result in a slightly different rounding method, but it should be and a program built with GCC gives exact same results).